### DISCUSSION TITLE
How to pop a page or multiple pages programatically.

### DISCUSSION THREAD

**Discussion by mjmon** [csells/go_router#124] [Q&A] [answered]
How to do this below for GoRouter

pop a page

`Navigator.pop();`

pop series of pages until a specific named route.
```
Navigator.of(context)
              .popUntil(ModalRoute.withName("/Page1"));
```

**Comment by csells** [comment, marked as answer]
You can use `Navigator.pop()` as usual when using go_router. However, instead of using `popUntil()` I recommend just using `context.go()` or `context.goNamed()`, which will build the stack of pages for you with the location/name you specify.

**Reply by mjmon** [reply]
Thank you for your response.

just a clarification about popping multiple pages.

so if I have `PageA` -> `PageB` -> `PageC` stack and am currently on `PageC`,
If I used `context.goNamed("PageA")`, it will pop `PageC` and `PageB` until `PageA`?

**Reply by jamesblasco** [reply]
Just to clarify, `context.go()` would not animate the views that are popped no? It just replaces the route stack with a new one with no animation?

**Reply by csells** [reply]
`context.go()` replaces the route stack using the animation appropriate for that, i.e. by default it will have the animation that matches the kind of app you're using, e.g. `MaterialApp`.

**Comment by csells** [comment]
No. Instead, context.go and goNamed will replace the entire stack of pages with whatever set is produced by the PageA route. Check out the docs on sub-routes for the details.

**Reply by mjmon** [reply]
got it,. thank you
